NEWS.md

```diff
@@ -1,5 +1,12 @@
 This file contains a summary of important user-visible changes.
 
+**Changes**
+
+- API: Sort, Term, Op, Grammar and datatype objects do not maintain a reference
```

Comment:
Actually, do we even need to mention NodeManager here at all?  The user shouldnt have to care about what NodeManager is.

Reply:
Well, yes, technically the only thing important is that we can now share NodeManager instances between Solvers.

Reply:
For us, however, it might be good to have these details added? Idk, don't have a strong opinion.

Reply:
Actually, we have always shared NodeManagers between Solvers, its only the dependency that has changed.   So the wording is still confusing to me.

I think I'd prefer not mentioning the internals in this note.

Reply:
Right, we have always shared NodeManagers between Solvers, but until #9020 we were not able to share Terms, Sorts, etc. between Solver instances. I agree, let's not mention internals, and only mention this.